### PR TITLE
refactor build ova

### PR DESCRIPTION
**Pull request by changev**

### CODE REVIEW

File: src/pipeline/rackhd/ova/build.sh [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+       export BUILD_STAGE=BUILD_FINAL
+  else
+       echo "Build from begining"
+       export BUILD_STAGE=BUILD_ALL
+  fi
+
+  export RACKHD_VERSION=$RACKHD_VERSION
+  #export end
+
+  ./HWIMO-BUILD
```
Comment (PengTian0) [inline src/pipeline/rackhd/ova/build.sh:53]: The script mixed up so much functions: build vagrant box, build vagrant ovf, build ova, build vmx.
It's hard to read.
I think it's better to implement a independent shell script build_ova.sh
And it only need parameters: os_ver, rackhd_version, rackhd_apt_repository, [cache_image]

File: src/pipeline/rackhd/ova/build.sh [hunk 2]
```diff
@@ -0,0 +1,137 @@
+#!/bin/bash -x
+
```
Comment (panpan0000) [inline src/pipeline/rackhd/ova/build.sh:2]: per @PengTian0 , we can split the original HWIMO-BUILD into 2 scripts:
build_vagrant.sh build_ova.sh  , and put them in rackhd/packer .  either in this PR or next is okay.

**Review by panpan0000 — APPROVED** [review]

File: src/pipeline/rackhd/ova/Build.groovy [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+              "RACKHD_COMMIT=${env.RACKHD_COMMIT}",
+              "RACKHD_VERSION=${env.RACKHD_VERSION}",
+              "IS_OFFICIAL_RELEASE=${env.IS_OFFICIAL_RELEASE}",
+              "OVA_CACHE_BUILD=${env.OVA_CACHE_BUILD}",
+              "OS_VER=${env.OS_VER}",
+              "BUILD_TYPE=vmware",
+              "ARTIFACTORY_URL=${env.ARTIFACTORY_URL}",
+              "STAGE_REPO_NAME=${env.STAGE_REPO_NAME}",
+              "DEB_COMPONENT=${env.DEB_COMPONENT}",
+              "DEB_DISTRIBUTION=trusty"]) {
```
Comment (PengTian0) [inline src/pipeline/rackhd/ova/Build.groovy:21]: Please use them as arguments of the function.
Or handle it in your next PR.

Comment (changev) [reply src/pipeline/rackhd/ova/Build.groovy:21]: I'm writing next PR

**Comment by PengTian0** [comment]
👍

**Comment by panpan0000** [comment]
👍

**Comment by panpan0000** [comment]
the next work is a separated build_ova.sh to replace HWIMO-BUILD

**Pull request merged**